### PR TITLE
use graphql url from env

### PR DESCRIPTION
**Pull request by tatchi**
Addresses #64 

Vite can [load variables](https://vitejs.dev/guide/env-and-mode.html#env-files) from an `.env` file defined at the root of the project (`/frontend` in our case).

This PR allows defining a variable `VITE_GRAPHQL_URL` within the `environments/development.env` or `environments/development.env` file. During the `docker-compose` workflow, an `.env` file that contains this variable is created in the `frontend` folder, so that `vite` can use it when bundling the app.

That's nice because we can define all the variables in one file for the `docker-compose` workflow. One drawback though is that outside of the `docker-compose` workflow, we would still have to manually add an `.env` file under the `/frontend` folder.

The other solution would be to use this manually created `.env` file for both scenarios. We would lose the benefit of having only one file where we define env variables for the `docker-compose` workflow, but that might be simpler...

What do you think?

### CODE REVIEW

File: frontend/Dockerfile [hunk 1]
```diff
@@ -10,6 +12,8 @@ RUN yarn install
 # add app
 COPY . /app/
 
+RUN echo "VITE_GRAPHQL_URL=${VITE_GRAPHQL_URL}" > /app/.env
```
Comment (tatchi) [inline frontend/Dockerfile:15]: Not sure if that's the best way of doing that... I first tried to define a `volumes` in the `docker-compose.yaml` file and somehow trying to use it here but I wasn't able to make it work.

File: environments/development.docker-compose.yaml [hunk 2]
```diff
@@ -21,7 +21,10 @@ services:
       HASURA_GRAPHQL_ENABLED_LOG_TYPES: startup, http-log, webhook-log, websocket-log, query-log
       HASURA_GRAPHQL_ADMIN_SECRET: ${OCAML_BENCH_GRAPHQL_KEY?required}
   frontend:
-    build: ../frontend
+    build: 
+      context: ../frontend
+      args:
+        VITE_GRAPHQL_URL: ${VITE_GRAPHQL_URL}
```
Comment (gs0510) [inline environments/development.docker-compose.yaml:27]: We should add the required tag here as well.

File: frontend/Dockerfile (hunk 1 above)
Comment (gs0510) [reply frontend/Dockerfile:15]: Since we'll need this in the root, I think it's not a bad idea.

**Comment by rizo** [comment]
@tatchi would it be useful to set the env var to `localhost` by default?

**Comment by tatchi** [comment]
> @tatchi would it be useful to set the env var to `localhost` by default?

Yes definitely, will create a PR :)

**Pull request merged**